tools/pylint-extensions/pylint-guidelines-checker/pylint_guidelines_checker.py

```diff
@@ -1985,6 +1977,165 @@ def visit_importfrom(self, node):
                         confidence=None,
                     )
 
+
+class TypePropertyNameTooLong(BaseChecker):
+    __implements__ = IAstroidChecker
+
+    """Rule to check that the character length of type and property names are not over X characters."""
+    name = "name-too-long"
+    priority = -1
+    msgs = {
+        "C4751": (
+            "Name is over standard character length of 40.",
+            "name-too-long",
+            "Only use names that are less than 40 characters."
+        ),
+    }
+
+    STANDARD_CHARACTER_LENGTH = 40
+
+    def visit_classdef(self,node):
+        """Visit every class and check that the
+         class name is within the character length limit."""
+
+        try:
+            if len(node.name) > self.STANDARD_CHARACTER_LENGTH:
+                self.add_message(
+                    msgid=f"name-too-long",
+                    node=node,
+                    confidence=None,
+                )
+        except:
+            pass
+
+
+    def visit_functiondef(self, node):
+        """Visit every function and check that the function and 
+        its variable names are within the character length limit."""
+
+        try:
+            if len(node.name) > self.STANDARD_CHARACTER_LENGTH:
+                self.add_message(
+                    msgid=f"name-too-long",
+                    node=node,
+                    confidence=None,
+                )
+
+            for i in node.body:
+                try:
+                    if len(i.name) > self.STANDARD_CHARACTER_LENGTH:
+                        self.add_message(
+                            msgid=f"name-too-long",
+                            node=i,
+                            confidence=None,
+                        )        
+                except:
+                    # Gets the names of ast.Assign statements
+                    for j in i.targets:
+                        if len(j.name) > self.STANDARD_CHARACTER_LENGTH:
+                            self.add_message(
+                                msgid=f"name-too-long",
+                                node=j,
+                                confidence=None,
+                            )          
+        except:
+            pass      
+
+    visit_asyncfunctiondef = visit_functiondef
+
+
+class DeleteOperationReturnStatement(BaseChecker):
```

Comment:
Thinking about this one... the delete* seems pretty straightforward because we can sniff out whether it's returning a non-None value. The begin_delete* is a bit tougher because I don't think it's straightforward to tell what the final result of the LROPoller is in code. In general, we want to make sure this isn't something like LROPoller[JobStatus]. I wonder if it would make sense to check the type annotation for the method instead.

Reply:
It is now checking node.returns which can be populated by the typehints